### ISSUE TITLE
[BUG/ISSUE] Error when running GHCP on GCloud VM (ubuntu) : grow_record_size() - exceeded maximum permitted buffer size.

### ISSUE THREAD

**Issue by lorenzocostantino** [geoschem/gchp#195]
### What institution are you from?
CREA (Centre for Research on Energy and Clean Air)

### Description of the problem
Hi, I am a new GC user.
I followed step-by-step Youtube videos and the main manual web page. 
Compilation is OK. Rundir created correctly. GC built with -DCMAKE_BUILD_TYPE=Debug.
All other debug flags are also on.

When I run the ./gchp executable in my C24-rundir 

------> When I run : mpirun -np 6 --oversubscribe ./gchp

The model starts running but very early I have the following error: 

[....]
   Mem/Swap Used (MB) at HISTMAPL_GenericInitialize=  9.269E+03  0.000E+00
        GENERIC: INFO: Finished the 'Initialize' stage of the gridded component 'HIST'
        GENERIC: INFO: Finished the 'Initialize' stage of the gridded component 'HIST'
        GENERIC: INFO: Finished the 'Initialize' stage of the gridded component 'HIST'
        GENERIC: INFO: Finished the 'Initialize' stage of the gridded component 'HIST'
        GENERIC: INFO: Finished the 'Initialize' stage of the gridded component 'HIST'
        GENERIC: INFO: Finished the 'Initialize' stage of the gridded component 'HIST'
        GENERIC: INFO: Started the  'Initialize' stage of the gridded component 'EXTDATA'
        GENERIC: INFO: Started the  'Initialize' stage of the gridded component 'EXTDATA'
        GENERIC: INFO: Started the  'Initialize' stage of the gridded component 'EXTDATA'
        GENERIC: INFO: Started the  'Initialize' stage of the gridded component 'EXTDATA'
        GENERIC: INFO: Started the  'Initialize' stage of the gridded component 'EXTDATA'
        GENERIC: INFO: Started the  'Initialize' stage of the gridded component 'EXTDATA'
                                                          Mem/Swap Used (MB) at EXTDATAMAPL_GenericInitialize=  9.269E+03  0.000E+00
STOP 1
 /home/lorenzo/GCHP/src/pFlogger/src/DynamicBuffer.F90          43
 DynamicBuffer::grow_record_size() - exceeded maximum permitted buffer size.
--------------------------------------------------------------------------
[...]

### GEOS-Chem version
Google Clopud VM, 6 cores, 48 GB Mermory. Space: 500 GB
Linux VERSION="20.04.3 LTS (Focal Fossa)"
GCHP version : 13.3.4
All dependencies have been installed with spack, as explained in the manual using the one-line command :
spack install --only dependencies gchp ^openmpi

### Description of code modifications
None

### Log files
No specific errors in log files. For instance at the beginning (4% of file length) of allPEs.log I see: 

[...]
0002: CAP.EXTDATA: ERROR: ExtData could not find appropriate file from file template ./ for side R
0005: CAP.EXTDATA: ERROR: ExtData could not find appropriate file from file template ./ for side R
0000: CAP.EXTDATA: ERROR: ExtData could not find appropriate file from file template ./ for side R
0000: MAPL.SHMEM: INFO: NumCores per Node = 6
0000: MAPL.SHMEM: INFO: NumNodes in use   = 1
0000: MAPL.SHMEM: INFO: Total PEs         = 6
0000: MAPL.SHMEM: INFO: NumNodes in use  = 1
0000: MAPL.GENERIC: INFO: Started the  'SetServices' stage of the gridded component 'GCHP'
0000: MAPL.GENERIC: INFO: Started the  'SetServices' stage of the gridded component 'GCHPctmEnv'
0000: MAPL.GENERIC: INFO: Finished the 'SetServices' stage of the gridded component 'GCHPctmEnv'
0000: MAPL.GENERIC: INFO: Started the  'SetServices' stage of the gridded component 'GCHPchem'
0000: MAPL.GENERIC: INFO: Finished the 'SetServices' stage of the gridded component 'GCHPchem'
0000: MAPL.GENERIC: INFO: Started the  'SetServices' stage of the gridded component 'DYNAMICS'
0000: MAPL.GENERIC: INFO: Finished the 'SetServices' stage of the gridded component 'DYNAMICS'
0000: MAPL.GENERIC: INFO: Finished the 'SetServices' stage of the gridded component 'GCHP'
0000: MAPL.GENERIC: INFO: Started the  'SetServices' stage of the gridded component 'HIST'
0000: MAPL.GENERIC: INFO: Finished the 'SetServices' stage of the gridded component 'HIST'
0000: MAPL.GENERIC: INFO: Started the  'SetServices' stage of the gridded component 'EXTDATA'
0000: MAPL.GENERIC: INFO: Finished the 'SetServices' stage of the gridded component 'EXTDATA'
0000: MAPL.GENERIC: INFO: Started the  'Initialize' stage of the gridded component 'GCHP'
0000: MAPL.GENERIC: INFO: Started the  'Initialize' stage of the gridded component 'GCHPctmEnv'
0000: MAPL.GENERIC: INFO: Finished the 'Initialize' stage of the gridded component 'GCHPctmEnv'
0000: MAPL.GENERIC: INFO: Started the  'Initialize' stage of the gridded component 'GCHPchem'
0000: MAPL.GENERIC: INFO: Finished the 'Initialize' stage of the gridded component 'GCHPchem'
0000: MAPL.GENERIC: INFO: Started the  'Initialize' stage of the gridded component 'DYNAMICS'
0000: MAPL.GENERIC: INFO: Finished the 'Initialize' stage of the gridded component 'DYNAMICS'
0000: MAPL.GENERIC: INFO: Finished the 'Initialize' stage of the gridded component 'GCHP'
0000: MAPL.GENERIC: INFO: Started the  'Initialize' stage of the gridded component 'HIST'
0000: MAPL.GENERIC: INFO: Finished the 'Initialize' stage of the gridded component 'HIST'
0000: MAPL.GENERIC: INFO: Started the  'Initialize' stage of the gridded component 'EXTDATA'
0000: MAPL.GENERIC: INFO: Finished the 'Initialize' stage of the gridded component 'EXTDATA'
0000: MAPL.GENERIC: INFO: Started the  'Run' stage of the gridded component 'EXTDATA'
0000: CAP.EXTDATA: ERROR: ExtData could not find appropriate file from file template ./ for side R
0002: CAP.EXTDATA: ERROR: ExtData could not find appropriate file from file template ./ for side R...
[...]

### Software versions
- Compilers (Intel or GNU, and version): gcc (Spack GCC) 9.3.0, GNU Fortran (Spack GCC) 9.3.0
- NetCDF version: netcdf-c@4.8.1, netcdf-fortran@4.5.3
- MPI version: mpirun (Open MPI) 4.1.2
- ESMF version: esmf@8.0.1

**Comment by LiamBindle** [comment]
@lorenzocostantino Thanks for opening an issue. The following is a common error message that means a file needed for the "right side" of a time interpolation (between files) is missing:
```
0000: CAP.EXTDATA: ERROR: ExtData could not find appropriate file from file template ./ for side R
0002: CAP.EXTDATA: ERROR: ExtData could not find appropriate file from file template ./ for side R...
```
Often, this happens when a meteorlogical file is missing. Note that you need meteorological data for at least one day before your simulation starts, and one day after your simulation ends. Could you check if you have meteorological data for the appropriate dates?

If you do, perhaps try setting the `CAP.EXTDATA` logger to `INFO` or `DEBUG` level. This should get the logger to print all the files it's trying to locate, which will hopefully identify the file that's missing.

---

I'm not positive, but the 
```
DynamicBuffer::grow_record_size() - exceeded maximum permitted buffer size.
```
error looks like a red herring. It seems like a runtime error that might be occuring when it's trying to dump the error message. Someone please correct me if I'm wrong.

**Comment by lorenzocostantino** [comment]
Dear @LiamBindle, thank you so much for your inputs, that really helped me to move forward.

Actually, the CAP.EXTDATA logger was already on DEBUG level. 
But you were right, I did not download the D+1 meteo file. So I did it.

At first, this didn't solve my problem. I was still facing the same error messages :
`0000: CAP.EXTDATA: ERROR: ExtData could not find appropriate file from file template ./ for side R  --> in allPEs.log`
`DynamicBuffer::grow_record_size() - exceeded maximum permitted buffer size. --> in the std output`

Then, I changed the CAP.EXTDATA logger from DEBUG to WARNING level, and I managed to avoid the error: 
`DynamicBuffer::grow_record_size() - exceeded maximum permitted buffer size`
So it seems that the DEBUG flag in CAP.EXTDATA caused this last issue (when meteo data are not present?)

Then, I realized (running into a further error) that I also needed to download the VOLCANO so2 data. 
Adding this last dataset, I've been able to run my first GCHP simulation.
